massa-bootstrap/src/establisher.rs

```diff
@@ -1,91 +1,57 @@
 // Copyright (c) 2022 MASSA LABS <info@massa.net>
-use async_trait::async_trait;
 use massa_time::MassaTime;
-use std::{io, net::SocketAddr};
+use std::{
+    io,
+    net::{SocketAddr, TcpListener, TcpStream},
+    time::Duration,
+};
 
 /// duplex connection
-pub type Duplex = tokio::net::TcpStream;
+pub trait Duplex:
```

Comment:
Having this as a trait is handy, but look into mocking the TcpStream directly with `mockall::mock!` or something similar